src/ngraph/pass/opset1_downgrade.cpp

```diff
@@ -0,0 +1,98 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/pass/opset1_downgrade.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/op/get_output_element.hpp"
+#include "ngraph/op/pad.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+#define NGRAPH_OP(a, b) a,
```

Comment:
Is there any way we could avoid duplicating this `OP_TYPEID` code yet again?

Reply:
Good point, but I'll create separate PR for this request ;)